src/saving/serializers/ThriveJsonConverter.cs

```diff
@@ -68,9 +68,20 @@ private ThriveJsonConverter(SaveContext context)
 
     public static ThriveJsonConverter Instance => InstanceValue;
 
-    public string SerializeObject(object o)
-    {
-        return PerformWithSettings(settings => JsonConvert.SerializeObject(o, Constants.SAVE_FORMATTING, settings));
+    /// <summary>
+    ///   Serializes the specified object to a JSON string using ThriveJsonConverter settings.
+    /// </summary>
+    /// <param name="o">Object to serialize</param>
+    /// <param name="type">
+    ///   <para>
+    ///     This is used so we could pass in String as the Type to force the serializer to write out the type name,
+    ///     to fix (presumably a bug) where the object's type name is sometimes not written with TypeNameHandling.Auto.
```

Comment:
Could you open a new issue with the investigation results, that would be like investigate why this is needed (especially debug the type name handling, perhaps even an explicit parameter to override TypeNameHandling per invocation)? And link that here.